aliasing/api/combat.py

```diff
@@ -67,6 +70,51 @@ def get_group(self, name):
             return SimpleGroup(group)
         return None
 
+    def set_metadata(self, k: str, v: str):
+        """
+        Assigns a metadata key to the passed value.
+        Maximum size of the metadata is 100k characters, key and item inclusive.
+
+        :param str k: The metadata key to set
+        :param str v: The metadata value to set
+
+        Example:
+
+        >>> set_metadata("Test", dump_json({"Status": ["Mario", 1, 2]}))
+        """
+        if sum(len(k + v) for k, v in self._combat._metadata.items()) > MAX_METADATA_SIZE:
+            raise ValueError
+        self._combat._metadata[k] = v
```

Comment:
```suggestion
        self._combat._metadata[str(k)] = str(v)
        if sum(len(k + v) for k, v in self._combat._metadata.items()) > MAX_METADATA_SIZE:
            raise ValueError("Combat metadata is too large")
```
- Make sure to do the assignment before size checking so it's actually checking the size of the resulting dict
- Cast key, value to str just in case
- ValueError needs a message